proposed/http-factory/http-factory.md

```diff
@@ -163,14 +170,17 @@ fwrite($resource, $body);
 Has the ability to create streams for uploaded files.
 
 ```php
-namespace Psr\Http\Message;
+namespace Psr\Http\Factory;
+
+use Psr\Http\Message\UploadedFileInterface;
 
 interface UploadedFileFactoryInterface
 {
     /**
      * Create a new uploaded file.
      *
-     * If a string is passed it is assumed to be a file path.
+     * If a string is used to create the file, a temporary resource will be
+     * created with the content of the string.
```

Comment:
@shadowhand — It would allow a process to create the stream for an uploaded file discovered in the request stream *before* creating the actual `UploadedFileInterface` instances. In particular, when I was working with react, I had to parse the request manually, because it's not tied to a SAPI, and doesn't automatically create temporary filehandles. This approach would allow `php://temp` streams to be used in those instances, which could then be used to seed the `UploadedFileInterface` instances.